### DISCUSSION TITLE
Github Copilot not giving inline suggestions in VScode

### DISCUSSION THREAD

**Discussion by emansyl** [community/community#80400] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I recently noticed that Github Copilot does not give inline suggestions in my code, and I don't know what would have changed to cause this. Copilot chat works fine, and when I open the completions panel, I see the output, but it does not appear in my editor. 

This is a release version of VS code and GitHub Copilot, and I am working with React/typescript.

I have tried a few different options, such as:

- Uninstalling Copilot
- Checking for updates with Vscode and Copilot
- I checked the settings and searched the forum for other similar topics, but none of the suggestions were helpful. 

Do you have any suggestions on what could be wrong?

**Comment by Kinnith7** [comment]
Did you ever find a solution to this? I have asked myself and support for this product seems to be non-existent.

**Reply by emansyl** [reply]
Nope I haven't been able to find a solution.

**Comment by zulrang** [comment]
This started happening to me as of yesterday.  Any ideas yet?

**Comment by mishmanners** [comment]
Hmmm this is an odd one. Sometimes it can be to do with pinging the server. For example GitHub Copilot chat might give you plenty of suggestions, but inline might not. Sometimes if you have a really big code base it can take a while.

Do you see the little GitHub Copilot icon in the bottom right? Does this spin when you type or add a new line?

Also remember GitHub Copilot chat and GitHub Copilot are two different extensions in VS Code. Let me know if you have both extensions installed.

**Reply by zulrang** [reply]
There is no icon in the bottom right - this is the first thing I noticed.

But all the UI stuff is there.  If I press `Ctrl+I` I get the popup for "Ask Copilot to generate code..." and if I tell it to do something, it "thinks" for a moment then returns nothing.

**Reply by emansyl** [reply]
In my case, there is an Icon at the bottom right, and the status says ready. I have both extensions installed, and Copilot chat works with no issue. However, inline suggestions never work, even on a new project.

**Reply by mishmanners** [reply]
@zulrang if you don't have the GitHub Copilot icon it means you only have the GitHub Copilot chat extension installed, and not GitHub Copilot. Install the GitHub Copilot extension and you'll have inline code suggestions.

As for @emansyl, let me put this question to the Copilot team and see what they come back with.

**Reply by zulrang** [reply]
It is installed.  I've also tried reinstalling.  On the extension page, it has the options to `Disable` or `Uninstall` and under "Runtime Status" it says:
```
Activation Event:onStartupFinished
Activation Time:61ms
```

In "Output" under "Extension Host (Remote)" there is a line:
```
2024-01-15 18:52:46.916 [info] ExtensionService#_doActivateExtension GitHub.copilot, startup: false, activationEvent: 'onStartupFinished'
```

This is Copilot.  Not Copilot Chat.  Both are installed.  Both are enabled.  

It was working yesterday.

**Reply by zulrang** [reply]
Update: I completely disabled Copilot Chat and rebooted.

Pressing `Ctrl+I` and giving it commands *sometimes* results in creating code to be inserted, but suggestions are *never* occurring on their own. 

Still no icon in the bottom right though.

**Reply by mishmanners** [reply]
Hmmm this is very interesting. It's odd that the icon isn't there. The icon should be showing if GitHub Copilot is installed. Even if it's not authenticated or not connected to the internet, then the icon is there, but with a strike through. What version of VS are you using?

**Reply by zulrang** [reply]
1.85.1
Looks like Copilot was updated today.

Another thing of note: it's working fine (icon and all) when editing local files, but not when using remote WSL.  Yes, it is installed on the remote as well (and has been reinstalled).  Yes, I've killed the remote processes.

It works on remote SSH though.  Just not WSL

**Reply by emansyl** [reply]
@mishmanners Thank you! Looking forward to hearing back. Let me know if I can provide any details that can be helpful.

**Reply by mishmanners** [reply, marked as answer]
Okay turns out there are two other places where GitHub Copilot inline might not work, but the rest of the functionality is there:
1. Settings --> Extensions --> Copilot

There should be a setting there to enable or disable auto complete:
![image](https://github.com/community/community/assets/36594527/93aa7355-3946-46ce-8108-a0f3d2f0db74)

and number 2.
The "depreciated" setting that shows up under Advanced:
![image (1)](https://github.com/community/community/assets/36594527/59ba5de6-07e4-46fc-8c6b-5978de4030a7)

Let me know who you go with these.

**Reply by emansyl** [reply]
Thank you so much! It was disabled in the advanced settings. I changed this, and it's now working!!

 However, I am unsure how this was disabled in the first place, and I am confident I didn't make this change myself.

**Reply by mishmanners** [reply]
Yassss I am so glad this worked for you 💪 let me know if you need anymore help.

**Comment by TigerYWang** [comment]
Hey It's not working again starting this morning

**Reply by TigerYWang** [reply]
ok, it's back now. It was off for like probably 2 hours this morning.

**Comment by JacobKissee** [comment]
Yeah, mine will also just stop working randomly and when I try to figure it out or ask about it I just get the basic check for updates. Make sure everything's enabled like I've obviously already tried that. It's not a problem on my end.

**Comment by NeonLexie** [comment]
mine is not working and every setting is turned on and I am not out of tokens

**Reply by Sledmine** [reply]
I went trough the same frustration with no solution until now, it appears that suggestions are not a json setting anymore (or at least I was not able to find it after fixing it lol), I was in a weird situation where my VS Code was a bit older and the extension changed but did not load all the options the new extension has...

I was able to restore suggestions by enabling this setting:

"chat.extensionUnification.enabled": true

AND updating to the latest VS Code / Copilot extension, then this appeared in the Github Copilot icon from the status bar:

<img width="560" height="242" alt="image" src="https://github.com/user-attachments/assets/c31332a9-c30e-4e26-882f-f8051ca95320" />

Hope that helps.
